### PR TITLE
feat: add withCredentials option (close #220)

### LINKED ISSUE #220: add withCredentials option
## Version

v3.3.0-a

## Development Environment

All

## Current Behavior

When using `net`, need to set the `withCredentials` option to handle `Access-Control-Allow-Credentials: true` header response.

## Expected Behavior

```js
// readData
grid.use('Net', {
  ...
  withCredentials: true
});

// createDate, updateData, deleteData, modifyData
net.request('createData', {
  withCredentials: true
});
```

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
APIs
```js
// readData
grid.use('Net', {
  ...
  withCredentials: true
});

// createDate, updateData, deleteData, modifyData
net.request('createData', {
  withCredentials: true
});
```
---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/addon/net.js [hunk 1]
```diff
@@ -66,6 +67,7 @@ var errorMessageMap = {
  *         readDataMethod: 'GET',
  *         perPage: 500,
  *         enableAjaxHistory: true,
+ *         withCredentials: true,
```
Comment (seonim-ryu) [inline src/js/addon/net.js]: @todo 예제에서 디폴트 값 `false`로 변경할 것

**Comment by jungeun-cho** [comment]
이견 없습니다.~

**Comment by jungeun-cho** [comment]
withCredentials 옵션 추가하는것이네요. 이견 없습니다.:)

File: src/js/addon/net.js [hunk 2]
```diff
@@ -489,6 +494,7 @@ var Net = View.extend(/** @lends module:addon/net.prototype */{
      *      @param {String} [options.modifiedOnly=true] - Whether the request param only contains modified rows
```
Comment (sohee-lee7) [inline src/js/addon/net.js]: 이번 수정사항은 아니지만, 타입이 잘못 지정되어져 있는 것 같습니다.

File: src/js/addon/net.js [hunk 3]
```diff
@@ -49,6 +49,7 @@ var errorMessageMap = {
  *          @param {string} [options.api.downloadExcelAll] - URL for download all data as an excel-file
  *      @param {number} [options.perPage=500] - The number of items to be shown in a page
  *      @param {boolean} [options.enableAjaxHistory=true] - Whether to use the browser history for the ajax requests
+ *      @param {boolean} [options.withCredentials=false] - Whether to set true for cross-domain requests if needed
```
Comment (kyuwoo-choi) [inline src/js/addon/net.js]: 음.. 위에것과 맞춘것 같기는 한데 아래 같은 표현은 어떨까요.

`use browser history for ajax requests if true`
`allow cross-domain requests if true`

**Comment by jinwoo-kim-nhn** [comment]
문제 없어보입니다.수고하셨습니다.

File: src/js/addon/net.js (hunk 2 above)
Comment (seonim-ryu) [reply src/js/addon/net.js]: 매의 눈! 👍 감사합니다ㅎㅎ

File: src/js/addon/net.js (hunk 3 above)
Comment (seonim-ryu) [reply src/js/addon/net.js]: 움? `browser history` -> `withCredentials` 이거죵?ㅎ

Comment (seonim-ryu) [reply src/js/addon/net.js]: @kyuwoo-choi 
이렇게 최종 수정하였습니다.
`use withCredentials flag of XMLHttpRequest for ajax requests if true`

**Pull request merged**